presto-main/src/test/java/com/facebook/presto/operator/aggregation/fixedhistogram/TestFixedDoubleBreakdownHistogram.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.fixedhistogram;
+
+import com.google.common.collect.Streams;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestFixedDoubleBreakdownHistogram
+{
+    @Test
+    public void testGetters()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        assertEquals(histogram.getBucketCount(), 200);
+        assertEquals(histogram.getMin(), 3.0);
+        assertEquals(histogram.getMax(), 4.0);
+    }
+
+    @Test
+    public void testIllegalBucketWeightCount()
+    {
+        try {
+            new FixedDoubleBreakdownHistogram(-200, 3.0, 4.0);
+            fail("exception expected");
+        }
+        catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("bucketCount"));
+        }
+    }
+
+    @Test
+    public void testIllegalMinMax()
+    {
+        try {
+            new FixedDoubleBreakdownHistogram(-200, 3.0, 3.0);
+            fail("exception expected");
+        }
+        catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("bucketCount"));
+        }
+    }
+
+    @Test
+    public void testBasicOps()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        histogram.add(3.1, 100.0);
+        histogram.add(3.8, 200.0);
+        histogram.add(3.1, 100.0);
+        assertEquals(
+                Streams.stream(histogram.iterator()).mapToDouble(c -> c.getWeight()).sum(),
+                300.0);
+    }
+
+    @Test
+    public void testEqualValuesDifferentWeights()
+    {
+        FixedDoubleBreakdownHistogram histogram =
+                new FixedDoubleBreakdownHistogram(200, 3.0, 4.0);
+
+        histogram.add(3.5, 0.2, 1);
+        histogram.add(3.5, 0.4, 1);
+        histogram.add(3.5, 0.3, 2);
+        Streams.stream(histogram.iterator()).forEach(c -> {
```

Comment:
nit: for-loop

Reply:
I don't exactly see why. Could you elaborate? Thanks!

Reply:
@atavory : For simple operations, stream and lambda-style coding makes code concise and more "functional programming". However, when the inner logic is complicated ,it defeat the purpose of using lambda. And plain for-loop is easy to read.

Plain for-loop is more straightforward and easier to maintain. 

However, in many cases it's also a personal preference. 

BTW: Maybe call the variable `bucketWeight` instead of `c` ?